app/src/main/java/com/nervousfish/nervousfish/modules/pairing/AndroidBluetoothHandler.java

```diff
@@ -288,7 +288,7 @@ public void run() {
             BluetoothSocket socket;
 
             // Listen to the server socket if we're not connected
-            while (mState != AndroidBluetoothHandler.State.connected) {
+            while (mState != AndroidBluetoothHandler.State.connected && serverSocket != null) {
```

Comment:
The socket here is (and should be) always null though? Now we never get past this condition. Also why change this in this pr?